cmd/kube-controller-manager/app/options/options.go

```diff
@@ -189,7 +183,7 @@ func NewKubeControllerManagerOptions() (*KubeControllerManagerOptions, error) {
 
 	s.Authentication.RemoteKubeConfigFileOptional = true
 	s.Authorization.RemoteKubeConfigFileOptional = true
-	s.Authorization.AlwaysAllowPaths = []string{"/healthz"}
+	s.Authorization.AlwaysAllowPaths = []string{"/healthz", "/metrics"}
```

Comment:
I wasn't expecting to expose metrics to all callers. What unauthenticated callers are fetching metrics today?

@deads2k @sttts, thoughts here?

Reply:
> What unauthenticated callers are fetching metrics today?

As far as I know, if a user fetch metrics from controller-manager using prometheus, it is almost certain that the request is unauthenticated.

Reply:
Prometheus itself is capable of fetching metrics from endpoints that require authentication. Currently, basic auth and bearer tokens are supported methods as can be seen in [`scrape_config` section of prometheus configuration](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config). In case of kubernetes, a config like `bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token` is used frequently to authenticate to endpoints. This behavior can be seen in one of the examples: https://github.com/prometheus/prometheus/blob/master/documentation/examples/prometheus-kubernetes.yml#L59-L87

Reply:
Afaik in OpenShift Prometheus fetches metrics with a service account aka a token via delegated authentication.

Reply:
I would also consider this a regression, even security relevant: there are many parties disabling insecure ports and expecting delegated authn for the secure port. This suddenly exposes metrics to everybody. We shouldn't do this.

Reply:
Got it, thanks for pointing it out.